codegenerator/cli/npm/envio/src/FetchState.res

```diff
@@ -1226,3 +1226,21 @@ let isReadyToEnterReorgThreshold = (
   } &&
   queue->Utils.Array.isEmpty
 }
+
+let hasBatchItem = ({queue, latestFullyFetchedBlock}: t) => {
+  switch queue->Utils.Array.last {
+  | Some(item) => item.blockNumber <= latestFullyFetchedBlock.blockNumber
```

Comment:
In what case is this false? Is it a defensive check?

Reply:
Yes, or wildcard also creates a new partition